aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -753,6 +753,31 @@ func resourceAviatrixTransitGateway() *schema.Resource {
 					},
 				},
 			},
+			"interface_mapping": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "List of interface names mapped to interface types and indices.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "Interface name (e.g., 'eth0', 'eth1').",
+						},
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							Description:  "Interface type (e.g., 'wan', 'mgmt').",
+							ValidateFunc: validation.StringInSlice([]string{"WAN", "MANAGEMENT"}, false),
+						},
+						"index": {
+							Type:        schema.TypeString,
```

Comment:
Why string? Could you specify TypeInt and validate >= 0?

Reply:
Since the map is declared as a part of the list, its only accepting either all string values or int values. Changing the index type to int throws an error while using interface_mapping. Could not find a work around for this so using a string type for index.